data/mods/ssb/abilities.ts

```diff
@@ -10,49 +99,287 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
+			this.add('-ability', source, 'Scyphozoa');
+			this.add('-clearallboost');
+			for (const pokemon of this.getAllActive()) {
+				pokemon.clearBoosts();
+				for (stat in source.boosts) {
+					if (source.boosts[stat] < 6 && !exclude.includes(stat)) {
+						stats.push(stat);
+					}
+				}
+				if (stats.length) {
+					const randomStat = this.sample(stats);
+					const boost: SparseBoostsTable = {};
+					boost[randomStat] = 1;
+					this.boost(boost, source, source);
+				}
+				if (pokemon.removeVolatile('substitute')) {
+					for (stat in source.boosts) {
+						if (source.boosts[stat] < 6 && !exclude.includes(stat)) {
+							stats.push(stat);
+						}
+					}
+					if (stats.length) {
+						const randomStat = this.sample(stats);
+						const boost: SparseBoostsTable = {};
+						boost[randomStat] = 1;
+						this.boost(boost, source, source);
+					}
+				}
+			}
+			const target = source.side.foe.active[0];
+
+			const removeAll = [
+				'reflect', 'lightscreen', 'auroraveil', 'safeguard', 'mist', 'gmaxsteelsurge', 'ferrofluid',
+				'spikes', 'toxicspikes', 'stealthrock', 'shiftingrocks', 'stickyweb',
+			];
+			const silentRemove = ['reflect', 'lightscreen', 'auroraveil', 'safeguard', 'mist'];
+			for (const sideCondition of removeAll) {
+				if (target.side.removeSideCondition(sideCondition)) {
+					if (!(silentRemove.includes(sideCondition))) {
+						this.add('-sideend', target.side, this.dex.getEffect(sideCondition).name, '[from] ability: Scyphozoa', '[of] ' + source);
+						for (stat in source.boosts) {
+							if (source.boosts[stat] < 6 && !exclude.includes(stat)) {
+								stats.push(stat);
+							}
+						}
+						if (stats.length) {
+							const randomStat = this.sample(stats);
+							const boost: SparseBoostsTable = {};
+							boost[randomStat] = 1;
+							this.boost(boost, source, source);
+						}
+					}
+				}
+				if (source.side.removeSideCondition(sideCondition)) {
+					if (!(silentRemove.includes(sideCondition))) {
+						this.add('-sideend', source.side, this.dex.getEffect(sideCondition).name, '[from] ability: Scyphozoa', '[of] ' + source);
+						for (stat in source.boosts) {
+							if (source.boosts[stat] < 6 && !exclude.includes(stat)) {
+								stats.push(stat);
+							}
+						}
+						if (stats.length) {
+							const randomStat = this.sample(stats);
+							const boost: SparseBoostsTable = {};
+							boost[randomStat] = 1;
+							this.boost(boost, source, source);
+						}
+					}
+				}
+			}
+			for (const clear in this.field.pseudoWeather) {
+				if (clear.endsWith('mod') || clear.endsWith('clause')) continue;
+				this.field.removePseudoWeather(clear);
+				for (stat in source.boosts) {
+					if (source.boosts[stat] < 6 && !exclude.includes(stat)) {
+						stats.push(stat);
+					}
+				}
+				if (stats.length) {
+					const randomStat = this.sample(stats);
+					const boost: SparseBoostsTable = {};
+					boost[randomStat] = 1;
+					this.boost(boost, source, source);
+				}
```

Comment:
Here's this code block again! A function would be nice, if that's possible (I'm not super familiar with how `abilities.ts` works).

Reply:
I'd rather not have a function for this, less functions for data files = better

Reply:
Id ratther just make Scyzophoziaisws just have one big for loop here to make a function